### ISSUE TITLE
Fix  calculation of currentFracHour

### ISSUE THREAD

**Issue by rs028** [AtChem/AtChem2#305]
Small error in the calculation of `currentFracHour` affects the calculation of the solar angles.

**Comment by spco** [comment]
This is documented at https://github.com/AtChem/AtChem2/pull/266#pullrequestreview-107428474.

**Comment by spco** [comment]
`currentDayOfYear` is calculated only at the start of each _outer_ timestep.

The callgraph for `calcZenith()` is 
```
mechanism_rates(t,y,p) -> getEnvVarsAtT(t) -> calcZenith(t, dec)
```
which means that the `t` in `calcZenith()` is the time at the _inner_ timestep. I'm not sure whether this causes the misalignment seen in the link above. By this I mean that it's possible that `t` has moved on into the next day within a single call to `CVODE()`, but this won't update `currentDayOfYear` as this is only updated once per _outer_ timestep.

From what I can see, I don't think this should make any difference to the calculation of quantities such as `lha` or `cosx`.

Have you seen an example where this definitely affects these quantities, or is it that the misalignment makes you suspect that the later quantities may be affected?

**Comment by spco** [comment]
Can I clarify that the time we use is always GMT, wherever the actual lat/long posiiton of the simulation is? This is what is being used in the calculation of LHA, but it too me a while to figure out. I can't see this documented anywhere, which would definitely be needed by users.

**Comment by rs028** [comment]
Yes it is always GMT. I think it is easier than dealing with local time (not to mention daylight saving) but I haven't really tested it for a range of lat/long.

**Comment by spco** [comment]
Closed by #331
